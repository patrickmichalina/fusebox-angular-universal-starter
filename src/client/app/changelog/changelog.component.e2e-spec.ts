import { baseUrl, browser } from '../../../../tools/test/jest.e2e-setup'

describe('Changelog Page', () => {
  it('should have title', async () => {
    expect.assertions(1)

    const page = browser.goto(`${baseUrl}/changelog`)
    const text = await page.evaluate(() => document.title)

    expect(text).toContain('Changelog - Fusebox Angular Universal Starter')
  })
})
