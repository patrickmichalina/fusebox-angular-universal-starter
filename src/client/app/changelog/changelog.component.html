<mat-card>
  <div [innerHTML]="changelog$ | async | MarkdownToHtml"></div>
</mat-card>