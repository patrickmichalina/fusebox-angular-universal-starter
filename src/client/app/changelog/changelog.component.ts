import { Observable } from 'rxjs/Observable'
import { HttpClient } from '@angular/common/http'
import { ChangeDetectionStrategy, Component } from '@angular/core'

@Component({
  selector: 'pm-changelog',
  templateUrl: './changelog.component.html',
  styleUrls: ['./changelog.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ChangelogComponent {
  readonly changelog$ = this.http.get('./changelog.md', { responseType: 'text' })
    .catch(a => Observable.of('Error loading changelog.md'))

  constructor(private http: HttpClient) { }
}
