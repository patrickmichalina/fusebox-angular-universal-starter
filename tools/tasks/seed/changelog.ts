import { Sparky } from 'fuse-box'
import { taskName } from '../../config/build.config'
import { BUILD_CONFIG } from '../../config/build.config'
import { createWriteStream } from 'fs'
const conventionalChangelog = require('conventional-changelog')

Sparky.task(taskName(__filename), () => {
  // heroku does not clone the git repo, so no meta data can be found :(
  if (process.env.HEROKU) return Promise.resolve()

  return conventionalChangelog({
    preset: 'angular',
    releaseCount: 0
  }).pipe(createWriteStream(`./${BUILD_CONFIG.outputDir}/CHANGELOG.md`))
})