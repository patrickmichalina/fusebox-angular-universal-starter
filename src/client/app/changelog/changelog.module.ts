import { ChangelogComponent } from './changelog.component'
import { ChangelogRoutingModule } from './changelog-routing.module'
import { NgModule } from '@angular/core'
import { SharedModule } from '../shared/shared.module'

@NgModule({
  imports: [ChangelogRoutingModule, SharedModule],
  declarations: [ChangelogComponent],
  exports: [ChangelogComponent]
})
export class ChangelogModule { }
