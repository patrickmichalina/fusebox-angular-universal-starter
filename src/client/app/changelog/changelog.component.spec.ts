import { ChangelogComponent } from './changelog.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { ChangelogModule } from './changelog.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

@Component({
  selector: 'test-component',
  template: '<pm-changelog></pm-changelog>'
})
class TestComponent { }

describe(ChangelogComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [AppTestingModule.forRoot(), ChangelogModule],
      declarations: [TestComponent]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(TestComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture).toMatchSnapshot()
  }))
})
