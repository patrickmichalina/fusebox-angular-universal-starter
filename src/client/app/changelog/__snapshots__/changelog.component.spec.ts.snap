// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangelogComponent should compile 1`] = `
<test-component>
  <pm-changelog>
    <mat-card
      class="mat-card"
    >
      <div />
    </mat-card>
  </pm-changelog>
</test-component>
`;
